src/NuGet.Clients/NuGet.PackageManagement.UI/Models/DetailControlModel.cs

```diff
@@ -132,21 +132,23 @@ protected virtual void Dispose(bool disposing)
             OnPropertyChanged(nameof(IconBitmap));
             OnPropertyChanged(nameof(PrefixReserved));
 
-            var getVersionsTask = searchResultPackage.GetVersionsAsync();
+            Task<IReadOnlyCollection<VersionInfoContextInfo>> getVersionsTask = searchResultPackage.GetVersionsAsync();
 
             _projectVersionConstraints = new List<ProjectVersionConstraint>();
 
             // Filter out projects that are not managed by NuGet.
-            var projects = _nugetProjects.Where(project => project.ProjectKind != NuGetProjectKind.ProjectK).ToArray();
+            IProjectContextInfo[] projects = _nugetProjects.Where(project =>
+                project.ProjectKind == NuGetProjectKind.PackagesConfig || project.ProjectKind == NuGetProjectKind.PackageReference).ToArray();
 
-            foreach (var project in projects)
+            IReadOnlyDictionary<string, IReadOnlyCollection<IPackageReferenceContextInfo>> projectsToInstalledPackages =
+                await projects.GetInstalledPackagesAsync(ServiceBroker, CancellationToken.None);
+
+            foreach (IProjectContextInfo project in projects)
             {
                 if (project.ProjectKind == NuGetProjectKind.PackagesConfig)
                 {
                     // cache allowed version range for each nuget project for current selected package
-                    IReadOnlyCollection<IPackageReferenceContextInfo> installedPackages = await project.GetInstalledPackagesAsync(
-                        ServiceBroker,
-                        CancellationToken.None);
+                    IReadOnlyCollection<IPackageReferenceContextInfo> installedPackages = projectsToInstalledPackages[project.ProjectId];
```

Comment:
I left feedback elsewhere, but it applies here as well.  `TryGetValue(...)` is safer, here and below.

Reply:
TBD based on other discussion...